modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -329,51 +321,100 @@ impl ServerModule for Wallet {
                     signature: val,
                 })
             })
-            .chain(stream::once(async { round_ci }))
             .collect::<Vec<WalletConsensusItem>>()
             .await;
 
-        // We force new epochs only if height changed, or we have peg-outs (more than
-        // just round_ci item)
-        if last_consensus_height < proposed_height || 1 < items.len() {
-            ConsensusProposal::Trigger(items)
-        } else {
-            ConsensusProposal::Contribute(items)
+        let block_height_proposal = self
+            .block_height()
+            .await
+            .saturating_sub(self.cfg.consensus.finality_delay);
+
+        if block_height_proposal != self.consensus_block_height(dbtx).await {
+            items.push(WalletConsensusItem::BlockHeight(block_height_proposal));
         }
+
+        let fee_rate_proposal = self.fee_rate().await;
+
+        if fee_rate_proposal != self.consensus_fee_rate(dbtx).await {
+            items.push(WalletConsensusItem::Feerate(fee_rate_proposal));
+        }
```

Comment:
The fee rate changing doesn't trigger an epoch, but that's not a problem (and indeed the old behavior too) but we'll have to think about how to propose consensus items in an AlephBFT world and how that would change the module interface.

E.g. `consensus_proposal` could return a stream and the Fedimint consensus just combines all these streams and sends their output to AlephBFT. But maybe you already had something else in mind and I forgot.

Reply:
Yes I do have something in mind: consensus_proposal can be simplified to always return all possible consensus items on every call; we call consensus_proposal every second or so (as only the transactions are latency critical) and send all of them to the broadcast. The broadcast then filters the redundant items within a session based on their consensus_hash. So in this case we would simply send the current feerate to the broadcast which would submit it only if it changed.